rust/agama-dbus-server/src/network/model.rs

```diff
@@ -81,6 +81,44 @@ impl NetworkState {
         Ok(())
     }
 
+    /// Updates a controller connection with a new one.
+    ///
+    /// It uses the `id` to decide which connection to update.
+    ///
+    /// Additionally, it registers the connection to be removed when the changes are applied.
+    pub fn update_controller_connection(
+        &mut self,
+        mut conn: Connection,
+        settings: HashMap<String, ControllerConfig>,
+    ) -> Result<(), NetworkStateError> {
+        let controller = conn.clone();
+
+        if let Connection::Bond(ref mut bond) = conn {
+            if let Some(ControllerConfig::Options(opts)) = settings.get("options") {
+                bond.set_options(opts.to_owned())?;
+            }
+
+            if let Some(ControllerConfig::Ports(ports)) = settings.get("ports") {
+                let new_ports: Vec<_> = ports
+                    .iter()
+                    .map(|port| {
+                        bond.find_port(port).cloned().unwrap_or_else(|| {
+                            ConnectionBuilder::new(port)
+                                .with_type(DeviceType::Ethernet)
+                                .with_interface(port)
+                                .with_controller(controller.id())
+                                .build()
```

Comment:
I'm struggle with this, when I think of a scenario which have a other DeviceType then Ethernet, e.g.:
```
    eth1 <-l-  eth1.11 -m-+-> bond0
                          |
                  eth2 -m-+  
```
How do you think we are going to implement this?

Reply:
We should obtain the device type from the system.

Reply:
what is, if it's also a virtual interface like:
```
vlan9 -m-+
         |-> bond0
 eth2 -m-+  
```
And we just about to create them?